Improve upgrade summary on smaller terminals

alimirjamali opened the pull request:
resolves: https://github.com/rpm-software-management/dnf5/issues/785
resolves: https://github.com/rpm-software-management/dnf5/issues/1785

m-blaha reviewed (commented):
Would you mind extending the commit message a bit? It would be really helpful if you could briefly describe what the change does and how it achieves that. Including a couple of examples showing the behavior of DNF5 before and after the change would also make it easier to understand the impact. Thanks a lot!

alimirjamali commented:
> Would you mind extending the commit message a bit? It would be really helpful if you could briefly describe what the change does and how it achieves that.

I updated the PR commit comments and added some comments to the code.

> Including a couple of examples showing the behavior of DNF5 before and after the change would also make it easier to understand the impact. Thanks a lot!

Is it OK if I make a PR to `ci-dnf-stack` and cross-link it to this PR for the examples? Since the output is going to be changed the unittests require update as well.

Thank you very much for the review.
